src/main/attached_database.cpp

```diff
@@ -54,6 +77,14 @@ AttachedDatabase::AttachedDatabase(DatabaseInstance &db, Catalog &catalog_p, Sto
 }
 
 AttachedDatabase::~AttachedDatabase() {
+
+	if (!IsSystem() && !catalog->InMemory()) {
+		auto &db_manager = db.GetDatabaseManager();
+		lock_guard<mutex> write_lock(db.GetDatabaseManager().GetDbPathsLock());
+		auto &db_paths = db_manager.GetDbPaths();
```

Comment:
Can we move all of this code into the actual database manager to avoid having to expose the internals?

Reply:
I tried as best as possible, but when rolling back from the `UndoBuffer`, I didn't know how to remove the path if not from within the destructor.

Reply:
Nvm, there is a good chance that I misunderstood your comment. I now removed this code with calls to `DatabaseManager::InsertDbPath` and `DatabaseManager::EraseDbPath`. I hope that's what you intended. :)